### ISSUE TITLE
Integration Tests Error: Exception: Unable to connect ElasticSearch server : Notice: Undefined index: websites

### ISSUE THREAD

**Issue by careys7** [Smile-SA/elasticsuite#738]
### Preconditions

Magento Version : 2.2.2

ElasticSuite Version : 2.5.0

Environment : Developer

Third party modules : No

### Steps to reproduce

1. Attempt to run Magento integration test framework (eg https://github.com/Smile-SA/elasticsuite/issues/692#issuecomment-358482686)

### Expected result

1. Magento can install as part of integration test bootstrap process 

### Actual result

1. The Integration test framework can't setup 

```
Exception: Unable to connect ElasticSearch server : Notice: Undefined index: websites in /var/www/current/src/vendor/magento/module-config/App/Config/Type/System.php on line 248


  [InvalidArgumentException]
  Parameter validation failed


Exception trace:
 () at /var/www/current/src/setup/src/Magento/Setup/Console/Command/InstallCommand.php:189
 Magento\Setup\Console\Command\InstallCommand->initialize() at /var/www/current/src/vendor/symfony/console/Command/Command.php:207
 Symfony\Component\Console\Command\Command->run() at /var/www/current/src/vendor/symfony/console/Application.php:843
 Symfony\Component\Console\Application->doRunCommand() at /var/www/current/src/vendor/symfony/console/Application.php:193
 Symfony\Component\Console\Application->doRun() at /var/www/current/src/vendor/magento/framework/Console/Cli.php:104
 Magento\Framework\Console\Cli->doRun() at /var/www/current/src/vendor/symfony/console/Application.php:117
 Symfony\Component\Console\Application->run() at /var/www/current/src/bin/magento:23

setup:install [--backend-frontname BACKEND-FRONTNAME] [--amqp-host AMQP-HOST] [--amqp-port AMQP-PORT] [--amqp-user AMQP-USER] [--amqp-password AMQP-PASSWORD] [--amqp-virtualhost AMQP-VIRTUALHOST] [--amqp-ssl AMQP-SSL] [--es-hosts ES-HOSTS] [--es-enable-ssl] [--es-user ES-USER] [--es-pass ES-PASS] [--key KEY] [--db-host DB-HOST] [--db-name DB-NAME] [--db-user DB-USER] [--db-engine DB-ENGINE] [--db-password DB-PASSWORD] [--db-prefix DB-PREFIX] [--db-model DB-MODEL] [--db-init-statements DB-INIT-STATEMENTS] [-s|--skip-db-validation] [--http-cache-hosts HTTP-CACHE-HOSTS] [--session-save SESSION-SAVE] [--session-save-redis-host SESSION-SAVE-REDIS-HOST] [--session-save-redis-port SESSION-SAVE-REDIS-PORT] [--session-save-redis-password SESSION-SAVE-REDIS-PASSWORD] [--session-save-redis-timeout SESSION-SAVE-REDIS-TIMEOUT] [--session-save-redis-persistent-id SESSION-SAVE-REDIS-PERSISTENT-ID] [--session-save-redis-db SESSION-SAVE-REDIS-DB] [--session-save-redis-compression-threshold SESSION-SAVE-REDIS-COMPRESSION-THRESHOLD] [--session-save-redis-compression-lib SESSION-SAVE-REDIS-COMPRESSION-LIB] [--session-save-redis-log-level SESSION-SAVE-REDIS-LOG-LEVEL] [--session-save-redis-max-concurrency SESSION-SAVE-REDIS-MAX-CONCURRENCY] [--session-save-redis-break-after-frontend SESSION-SAVE-REDIS-BREAK-AFTER-FRONTEND] [--session-save-redis-break-after-adminhtml SESSION-SAVE-REDIS-BREAK-AFTER-ADMINHTML] [--session-save-redis-first-lifetime SESSION-SAVE-REDIS-FIRST-LIFETIME] [--session-save-redis-bot-first-lifetime SESSION-SAVE-REDIS-BOT-FIRST-LIFETIME] [--session-save-redis-bot-lifetime SESSION-SAVE-REDIS-BOT-LIFETIME] [--session-save-redis-disable-locking SESSION-SAVE-REDIS-DISABLE-LOCKING] [--session-save-redis-min-lifetime SESSION-SAVE-REDIS-MIN-LIFETIME] [--session-save-redis-max-lifetime SESSION-SAVE-REDIS-MAX-LIFETIME] [--cache-backend CACHE-BACKEND] [--cache-backend-redis-server CACHE-BACKEND-REDIS-SERVER] [--cache-backend-redis-db CACHE-BACKEND-REDIS-DB] [--cache-backend-redis-port CACHE-BACKEND-REDIS-PORT] [--page-cache PAGE-CACHE] [--page-cache-redis-server PAGE-CACHE-REDIS-SERVER] [--page-cache-redis-db PAGE-CACHE-REDIS-DB] [--page-cache-redis-port PAGE-CACHE-REDIS-PORT] [--page-cache-redis-compress-data PAGE-CACHE-REDIS-COMPRESS-DATA] [--base-url BASE-URL] [--language LANGUAGE] [--timezone TIMEZONE] [--currency CURRENCY] [--use-rewrites USE-REWRITES] [--use-secure USE-SECURE] [--base-url-secure BASE-URL-SECURE] [--use-secure-admin USE-SECURE-ADMIN] [--admin-use-security-key ADMIN-USE-SECURITY-KEY] [--admin-user ADMIN-USER] [--admin-password ADMIN-PASSWORD] [--admin-email ADMIN-EMAIL] [--admin-firstname ADMIN-FIRSTNAME] [--admin-lastname ADMIN-LASTNAME] [--cleanup-database] [--sales-order-increment-prefix SALES-ORDER-INCREMENT-PREFIX] [--use-sample-data] [-i|--interactive] [--magento-init-params MAGENTO-INIT-PARAMS]
 in /var/www/current/src/vendor/magento/framework/Shell.php:61
Stack trace:
#0 /var/www/current/src/dev/tests/integration/framework/Magento/TestFramework/Application.php(466): Magento\Framework\Shell->execute('/usr/bin/php -f...', Array)
#1 /var/www/current/src/dev/tests/integration/framework/bootstrap.php(77): Magento\TestFramework\Application->install(true)
#2 /var/www/current/src/vendor/phpunit/phpunit/src/Util/Fileloader.php(61): include_once('/var/www/curren...')
#3 /var/www/current/src/vendor/phpunit/phpunit/src/Util/Fileloader.php(45): PHPUnit\Util\Fileloader::load('/var/www/curren...')
#4 /var/www/current/src/vendor/phpunit/phpunit/src/TextUI/Command.php(915): PHPUnit\Util\Fileloader::checkAndLoad('/var/www/curren...')
#5 /var/www/current/src/vendor/phpunit/phpunit/src/TextUI/Command.php(737): PHPUnit\TextUI\Command->handleBootstrap('/var/www/curren...')
#6 /var/www/current/src/vendor/phpunit/phpunit/src/TextUI/Command.php(152): PHPUnit\TextUI\Command->handleArguments(Array)
#7 /var/www/current/src/vendor/phpunit/phpunit/src/TextUI/Command.php(141): PHPUnit\TextUI\Command->run(Array, true)
#8 /var/www/current/src/vendor/phpunit/phpunit/phpunit(53): PHPUnit\TextUI\Command::main()
#9 {main}

Next Magento\Framework\Exception\LocalizedException: Command returned non-zero exit code:
`/usr/bin/php -f '/var/www/current/src/bin/magento' setup:install -vvv --db-host='mysql' --db-user='root' --db-password='root' --db-name='magento_integration_tests' --backend-frontname='backend' --admin-user='user' --admin-password='password1' --admin-email='admin@example.com' --admin-firstname='firstname' --admin-lastname='lastname' --es-hosts='elasticsearch:9200' --magento-init-params='MAGE_DIRS[etc][path]=/var/www/current/src/dev/tests/integration/tmp/sandbox-0-6ff55500c9cc215a52a9e902f42c977c/etc&MAGE_DIRS[var][path]=/var/www/current/src/dev/tests/integration/tmp/sandbox-0-6ff55500c9cc215a52a9e902f42c977c/var&MAGE_DIRS[media][path]=/var/www/current/src/dev/tests/integration/tmp/sandbox-0-6ff55500c9cc215a52a9e902f42c977c/pub/media&MAGE_DIRS[static][path]=/var/www/current/src/dev/tests/integration/tmp/sandbox-0-6ff55500c9cc215a52a9e902f42c977c/pub/static&MAGE_DIRS[view_preprocessed][path]=/var/www/current/src/dev/tests/integration/tmp/sandbox-0-6ff55500c9cc215a52a9e902f42c977c/var/view_preprocessed/pub/static&MAGE_DIRS[code][path]=/var/www/current/src/dev/tests/integration/tmp/sandbox-0-6ff55500c9cc215a52a9e902f42c977c/generated/code&MAGE_DIRS[cache][path]=/var/www/current/src/dev/tests/integration/tmp/sandbox-0-6ff55500c9cc215a52a9e902f42c977c/var/cache&MAGE_DIRS[log][path]=/var/www/current/src/dev/tests/integration/tmp/sandbox-0-6ff55500c9cc215a52a9e902f42c977c/var/log&MAGE_DIRS[session][path]=/var/www/current/src/dev/tests/integration/tmp/sandbox-0-6ff55500c9cc215a52a9e902f42c977c/var/session&MAGE_DIRS[tmp][path]=/var/www/current/src/dev/tests/integration/tmp/sandbox-0-6ff55500c9cc215a52a9e902f42c977c/var/tmp&MAGE_DIRS[upload][path]=/var/www/current/src/dev/tests/integration/tmp/sandbox-0-6ff55500c9cc215a52a9e902f42c977c/var/upload&MAGE_DIRS[pub][path]=/var/www/current/src/dev/tests/integration/tmp/sandbox-0-6ff55500c9cc215a52a9e902f42c977c/pub&MAGE_MODE=developer' 2>&1` in /var/www/current/src/vendor/magento/framework/Shell.php:62
Stack trace:
#0 /var/www/current/src/dev/tests/integration/framework/Magento/TestFramework/Application.php(466): Magento\Framework\Shell->execute('/usr/bin/php -f...', Array)
#1 /var/www/current/src/dev/tests/integration/framework/bootstrap.php(77): Magento\TestFramework\Application->install(true)
#2 /var/www/current/src/vendor/phpunit/phpunit/src/Util/Fileloader.php(61): include_once('/var/www/curren...')
#3 /var/www/current/src/vendor/phpunit/phpunit/src/Util/Fileloader.php(45): PHPUnit\Util\Fileloader::load('/var/www/curren...')
#4 /var/www/current/src/vendor/phpunit/phpunit/src/TextUI/Command.php(915): PHPUnit\Util\Fileloader::checkAndLoad('/var/www/curren...')
#5 /var/www/current/src/vendor/phpunit/phpunit/src/TextUI/Command.php(737): PHPUnit\TextUI\Command->handleBootstrap('/var/www/curren...')
#6 /var/www/current/src/vendor/phpunit/phpunit/src/TextUI/Command.php(152): PHPUnit\TextUI\Command->handleArguments(Array)
#7 /var/www/current/src/vendor/phpunit/phpunit/src/TextUI/Command.php(141): PHPUnit\TextUI\Command->run(Array, true)
#8 /var/www/current/src/vendor/phpunit/phpunit/phpunit(53): PHPUnit\TextUI\Command::main()
#9 {main}
```

This is using the new install option added in https://github.com/Smile-SA/elasticsuite/pull/716/

It looks like the error is thrown inside ClientFactory::create() 

```php
try {
  $clientsOptions = array_filter($this->getClientOptions($options, $deploymentConfig));
  $this->clientFactory->create(['options' => $clientsOptions])->info();
} catch (\Exception $e) {
 $errors[] = "Unable to connect ElasticSearch server : {$e->getMessage()}";
}
```

Possibly avoiding the configuration object during setup initialisation will avoid the error - i.e. https://github.com/Smile-SA/elasticsuite/pull/693/

**Comment by afoucret** [comment]
Hi @careys7,

I will not have much time this week to work on this. Maybe can you try to propose some code that fix this one. I will be happy to review

**Comment by careys7** [comment]
Thanks @afoucret. Will try and send something in the next day or two.

**Comment by careys7** [comment]
@afoucret - see #745

I think the best approach is to avoid trying to initialise ScopeConfigInterface, so I have made a separate configuration implementation just for setup validation. 

This seemed like the cleanest way to avoid the ScopeConfigInterface without encapsulating the ES builder, exposing this to a new validator class, and updating the client to interface with an injected ES builder encapsulation.

**Comment by afoucret** [comment]
PR #769 will fix this one.
Will be released with 2.5.2

I close and propose you to track in #763 which is basically the same.

BR
